src/primitives/rectangle/mod.rs

```diff
@@ -239,6 +243,162 @@ impl Rectangle {
+    ///
+    /// By combining this method with [`columns`] it is possible to iterator over all pixels inside
+    /// the rectangle. This can be more flexible than using the [`points`] iterator, for example,
+    /// if a different iteration order is required or some operations should be called once per row.
+    ///
+    /// ```
+    /// use embedded_graphics::{prelude::*, primitives::Rectangle};
+    ///
+    /// let rect = Rectangle::new(Point::new(10, 20), Size::new(3, 4));
+    ///
+    /// // Iterate over the y coordinates of the rows in reverse order.
+    /// for y in rect.rows().rev() {
+    ///     for x in rect.columns() {
+    ///         // use x, y coordinates
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// [`columns`]: #method.columns
+    /// [`points`]: ../trait.Primitive.html#tymethod.points
+    pub fn rows(&self) -> Range<i32> {
+        self.top_left.y
+            ..self
+                .top_left
+                .y
+                .saturating_add(self.size.height.saturating_cast())
+    }
+
+    /// Returns the range of X coordinates in this rectangle.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use embedded_graphics::{prelude::*, primitives::Rectangle};
+    ///
+    /// let rect = Rectangle::new(Point::new(10, 20), Size::new(3, 4));
+    ///
+    /// assert_eq!(rect.columns(), 10..13);
+    /// ```
+    ///
+    /// By combining this method with [`rows`] it is possible to iterator over all pixels inside
+    /// the rectangle. This can be more flexible than using the [`points`] iterator, for example,
+    /// if a different iteration order is required or some operations should be called once per row.
+    ///
+    /// ```
+    /// use embedded_graphics::{prelude::*, primitives::Rectangle};
+    ///
+    /// let rect = Rectangle::new(Point::new(10, 20), Size::new(3, 4));
+    ///
+    /// // Iterate over all points starting from the top right corner and advancing downwards.
+    /// for x in rect.columns().rev() {
+    ///     for y in rect.rows() {
+    ///         // use x, y coordinates
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// [`rows`]: #method.rows
+    /// [`points`]: ../trait.Primitive.html#tymethod.points
+    pub fn columns(&self) -> Range<i32> {
+        self.top_left.x
+            ..self
+                .top_left
+                .x
+                .saturating_add(self.size.width.saturating_cast())
+    }
+
+    /// Returns `true` is the rectangle is zero sized.
+    ///
+    /// A rectangle is zero sized if the width or height are zero.
+    ///
+    /// # Examples
+    /// ```
+    /// use embedded_graphics::{prelude::*, primitives::Rectangle};
+    ///
+    /// let rect = Rectangle::new(Point::new(10, 20), Size::new(10, 20));
+    /// assert_eq!(rect.is_zero_sized(), false);
+    ///
+    /// let rect = Rectangle::new(Point::new(10, 20), Size::zero());
+    /// assert_eq!(rect.is_zero_sized(), true);
+    /// ```
+    pub fn is_zero_sized(&self) -> bool {
```

Comment:
Would you mind adding an MSRV comment to this method so we can change it to `const` when upgrading to at least 1.46.0? I know it's a bit of noise, but it makes grepping the codebase for reasons to upgrade a lot easier:

 ```suggestion
 // MSRV: Add const when upgrading to at least 1.46.0
    pub fn is_zero_sized(&self) -> bool {
```

Conditionals were [stabilised in 1.46](https://github.com/rust-lang/rust/blob/master/RELEASES.md#version-1460-2020-08-27).